src/main/java/net/fabricmc/loom/configuration/providers/mappings/tiny/MappingsMerger.java

```diff
@@ -70,8 +81,27 @@ public static void mergeAndSaveMappings(Path from, Path out, IntermediateMapping
 		try (var writer = new Tiny2FileWriter(Files.newBufferedWriter(out, StandardCharsets.UTF_8), false)) {
 			officialTree.accept(writer);
 		}
+	}
 
-		LOGGER.info(":merged mappings in " + stopwatch.stop());
+	private static void legacyMergeAndSaveMappings(Path from, Path out, IntermediateMappingsService intermediateMappingsService) throws IOException {
+		MemoryMappingTree intermediaryTree = new MemoryMappingTree();
+		intermediateMappingsService.getMemoryMappingTree().accept(intermediaryTree);
```

Comment:
Question: Whats the intended source namespace for this tree? In the normal path `MappingSourceNsSwitch` is used to ensure that intermediary is used.

Reply:
This assumes the intermediate mappings for <1.3 will be provided as (intermediary) -> [clientOfficial, serverOfficial]. This is done because intermediary is the only complete namespace. The other two have 'holes' which is problematic for field and method descriptors.

Reply:
That makes sense, it might be worth checking that the source ns of the MemoryMappingTree is intermediary after redaing in the mappings.